### PR TITLE
Draft: Ci/actions

### PR DESCRIPTION
**Pull request by y-eight**
## Motivation

Adding several actions:
- SAST
- Build image and container scan
- Build image and push to registry

## Changes

For additional information look at the commits.

## Tests done

- actions tested in branch

## ToDo
- [ ] binary build

### CODE REVIEW

**Comment by y-eight** [comment]
Pls squash and merge due to multiple test commits

**Comment by y-eight** [comment]
SAST is fail, and needs to be fixed in a separate PR

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM cgr.dev/chainguard/go:latest as gobuild
```
Comment (puffitos) [inline Dockerfile:1]: Maybe use the go1.21 image equivalent, no need for latest :)

File: .github/workflows/test_sast.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    env:
+      GO111MODULE: on
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (lvlcn-t) [inline .github/workflows/test_sast.yml:18]: use checkout v4

File: Dockerfile (hunk 1 above)
Comment (y-eight) [reply Dockerfile]: dynamically using version from gomod

**Pull request closed without merging**